Mathlib/LinearAlgebra/LinearProgramming.lean

```diff
@@ -0,0 +1,76 @@
+/-
+Copyright (c) 2023 Martin Dvorak. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Martin Dvorak, Antoine Chambert-Loir
+-/
+import Mathlib.Analysis.Convex.Cone.Pointed
+import Mathlib.Algebra.Order.Group.Defs
+
+/-!
+
+# Linear programming
+
+TODO
+
+-/
+
+/-- Typically `M` is `ℝ^m` and `N` is `ℝ^n` -/
+structure LinearProgram (R V W M N : Type*) [OrderedRing R]
+    [AddCommGroup V] [Module R V] [AddTorsor V M]
+    [AddCommGroup W] [Module R W] [AddTorsor W N]
+    where
```

Comment:
My original suggestion was only that `M` should be affine, not `N`. I don't really understand what `N` is (you didn't write any documentation!), but `M` is natural space in which the problem is set up, which may as well be affine

Reply:
I am familiar with finite-dimensional setting. In finite-dimensional setting, where LP is typically represented by a matrix, my `M` represents variables and my `N` represents conditions.

Reply:
For me personally, modelling LP in my head is easier in terms of "searching for a vector" than "searching for a point". I understand that "searching for a point" has certain appeal and I am therefore willing to rephrase the definitions to the affine setting; however, if `M` should be be affine, I will make `N` affine as well, so that the dual of LP is still kinda LP.